nilearn/tests/test_signal.py

```diff
@@ -1075,8 +1076,55 @@ def test_handle_scrubbed_volumes():
         interpolated_confounds[sample_mask, :], confounds[sample_mask, :]
     )
 
-    scrubbed_signals, scrubbed_confounds = nisignal._handle_scrubbed_volumes(
-        signals, confounds, sample_mask, "cosine", 2.5
+    (
+        scrubbed_signals,
+        scrubbed_confounds,
+        sample_mask,
+    ) = nisignal._handle_scrubbed_volumes(
+        signals, confounds, sample_mask, "cosine", 2.5, True
     )
     np.testing.assert_equal(scrubbed_signals, signals[sample_mask, :])
     np.testing.assert_equal(scrubbed_confounds, confounds[sample_mask, :])
+
+
+def test_handle_scrubbed_volumes_extrapolation():
+    """Check interpolation of signals with and without extrapolation."""
+    signals, _, confounds = generate_signals(
+        n_features=11, n_confounds=5, length=40
+    )
+
+    censored_samples = 5
+    sample_mask = np.arange(signals.shape[0])
+    scrub_index = np.arange(censored_samples)
```

Comment:
I would censor a few more parts in the signal, since there's part of the code manipulating the index of the new sample mask.
```suggestion
    scrub_index = np.arange(censored_samples)
    scrub_index = np.concatenate((censored_samples, np.array([20, 21, 32])))
```

Reply:
done, see 05014cc